SecondSemester/Routers/NetworkGraph.cs

```diff
@@ -0,0 +1,193 @@
+// <copyright file="NetworkGraph.cs" company="Elena Makarova">
+// Copyright (c) Elena Makarova. All rights reserved.
+// </copyright>
+
+namespace Routers;
+
+/// <license>
+/// https://github.com/whflf/homework/blob/main/LICENSE .
+/// </license>
+/// <summary>
+/// Represents a network graph and provides methods to generate router configurations.
+/// </summary>
+/// <remarks>
+/// This class can read a network topology from an input file, build a graph representation, and generate router configurations based on a maximum spanning tree algorithm.
+/// </remarks>
+public class NetworkGraph
+{
+    private readonly List<List<Tuple<int, int>>> routersWithNeighbours = [];
```

Comment:
Можно проще, явно именуя элементы кортежа
```suggestion
    private readonly List<List(int Router, int Neighbour)>> routersWithNeighbours = [];
```